examples/addr2line.rs

```diff
@@ -169,9 +204,18 @@ fn main() {
     } else {
         None
     };
+    let mut load_sup_section = |id: gimli::SectionId| -> Result<_> {
+        if let Some(ref sup_object) = sup_object {
+            load_file_section(id, &sup_object, endian, &arena_data)
+        } else {
+            Ok(gimli::EndianSlice::new(&[][..], endian))
+        }
+    };
 
     let symbols = object.symbol_map();
-    let ctx = Context::new_with_sup(object, sup_object.as_ref()).unwrap();
+    let dwarf = gimli::Dwarf::load(&mut load_section, &mut load_sup_section).unwrap();
+
+    let ctx = Context::from_dwarf(dwarf).unwrap();
```

Comment:
Please use `from_dwarf_with_sup`.

Reply:
I saw this & this part of the API wasn't clear to me (didn't seem to gel in a coherent way). Specifically, `Dwarf::Load` takes in a non-optional `sup` itself. If I did a `Dwarf::load` on the sup, what would be the sup I provide there? Is the sup section loader in `Dwarf::load` something else?

Reply:
Yeah this part of the API isn't great, and I've been playing with a different way of doing it, but I'm still not sure.

For now, you need to call `Dwarf::load` twice, the same as what `Context::new_with_sup` does.
- call Dwarf::load with `load_section` and `load_sup_section`. For this call, only `.debug_str` will be loaded from the supplementary object file.
- call Dwarf::load with `load_sup_section` and a closure that always returns empty sections.

This does mean that `.debug_str` will be loaded twice from the supplementary object file.